### PR TITLE
Separate Runner for Linting.

### LINKED ISSUE #303: Create a initial Runner for Lint Checks
## Description
Currently all the runners do Linting checks separately which leads to cache error on pre-commit.We would create a initial runner which does the lint checks.

## How to Reproduce
Run the CI on a commit.
https://github.com/OpenMined/SyMPC/runs/3301868996?check_suite_focus=true

## Expected Behavior
All the CI Tests should pass.

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Assigning a  separate runner for linting , to solve cache error ,resolves #303 
https://github.com/OpenMined/SyMPC/runs/3301868665?check_suite_focus=true
.

## How has this been tested?
CI

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by kamathhrishi — APPROVED** [review]
Looks good to me

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -37,49 +37,93 @@ jobs:
...
           bandit -r src
           safety check
 
+
+  python-tests:
+    needs: [python-linting]
+    strategy:
+      max-parallel: 2
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (gmuraru) [inline .github/workflows/tests.yml:84]: Q: Can't we have only ubuntu-latest here also?

**Review by gmuraru — APPROVED** [review]
Only one Q, in rest LGTM!

File: .github/workflows/tests.yml (hunk 1 above)
Comment (rasswanth-s) [reply .github/workflows/tests.yml:84]: For the normal tests, we would have only ubuntu?

Comment (gmuraru) [reply .github/workflows/tests.yml:84]: Ahh...those are the actual tests.
I read `needs: [python-linting]` as `those are the python linting tests`

**Pull request merged**